lib/spaceship/portal/portal_client.rb

```diff
@@ -376,7 +376,17 @@ def download_provisioning_profile(profile_id, mac: false)
       end
     end
 
+    def get_provisioning_profile(profile_id, mac: false)
+      r = request(:post, "account/#{platform_slug(mac)}/profile/getProvisioningProfile.action", {
+        teamId: team_id,
+        provisioningProfileId: profile_id
+      })
+      parse_response(r)
+    end
+
     def delete_provisioning_profile!(profile_id, mac: false)
+      get_provisioning_profile(profile_id, mac: mac)
```

Comment:
Instead, you can just add the following here

```ruby
if csrf_tokens.count == 0
        # If we directly create a new profile without querying anything before
        # we don't have a valid csrf token, that's why we have to do at least one request
        provisioning_profiles
      end
```
This is already used in the `create_provisioning_profile!` method